src/PrestaShopBundle/Controller/Admin/Sell/Customer/CustomerController.php

```diff
@@ -564,6 +593,25 @@ public function toggleNewsletterSubscriptionAction($customerId)
         return $this->redirectToRoute('admin_customers_index');
     }
 
+    /**
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_customers_index",
+     *     message="You do not have permission to edit this."
+     * )
+     *
+     * @param int $customerId
+     *
+     * @return JsonResponse
+     */
+    public function toggleNewsletterSubscriptionThenReturnJsonAction($customerId)
+    {
+        return $this->performActionThenReturnJsonResponse(
+            'toggleCustomerNewsletterSubscription',
+            (int) $customerId
+        );
+    }
+
```

Comment:
```suggestion
    public function toggleNewsletterSubscriptionThenReturnJsonAction(int $customerId)
    {
        return $this->performActionThenReturnJsonResponse(
            'toggleCustomerNewsletterSubscription',
            $customerId
        );
    }

```

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/25115/commits/5aabb5bb639e79844a62601873be667ab4bb33ab